AWSIoT/AWSIoTDataManager.h

```diff
@@ -686,6 +695,40 @@ DEPRECATED_MSG_ATTRIBUTE("Use `updateUserMetaData` for updating the user meta da
          extendedCallback:(AWSIoTMQTTExtendedNewMessageBlock)callback
               ackCallback:(AWSIoTMQTTAckBlock)ackCallback;
 
+/**
+ Subscribes to a topic at a specific QoS level
+
+ @param topic The Topic to subscribe to.
+
+ @param qos Specifies the QoS Level of the subscription: AWSIoTMQTTQoSAtMostOnce or AWSIoTMQTTQoSAtLeastOnce
+
+ @param callback Reference to AWSIoTMQTTFullMessageBlock. When new message is received the callback will be invoked.
+
+ @return Boolean value indicating success or failure.
+
+ */
+- (BOOL)subscribeToTopic:(NSString *)topic
+                     QoS:(AWSIoTMQTTQoS)qos
+            fullCallback:(AWSIoTMQTTFullMessageBlock)callback;
+
+/**
+ Subscribes to a topic at a specific QoS level
+
+ @param topic The Topic to subscribe to.
+
+ @param qos Specifies the QoS Level of the subscription: AWSIoTMQTTQoSAtMostOnce or AWSIoTMQTTQoSAtLeastOnce
+
+ @param callback Reference to AWSIoTMQTTFullMessageBlock. When new message is received the callback will be invoked.
+
+ @param ackCallback the callback for ack if QoS > 0.
+
+ @return Boolean value indicating success or failure.
+
+ */
+- (BOOL)subscribeToTopic:(NSString *)topic
+                     QoS:(AWSIoTMQTTQoS)qos
+            fullCallback:(AWSIoTMQTTFullMessageBlock)callback
+             ackCallback:(AWSIoTMQTTAckBlock)ackCallback;
```

Comment:
I am making updates on the header to make the use of `ackCallback` correct. The doc comments has this line which is important.

"the callback for ack if QoS > 0"

It will be checked with the code below and throw an exception if it is defined if QoS is 0. This means for Swift there is no way to call these methods with a nil callback without the parameter being marked as `nullable` which it has not been for the pre-existing methods. The next commit will correct the header for nullability which will be a breaking change for Swift. It will require a minor version bump.

```objc
    if (qos == 0 && ackCallback != nil) {
        [NSException raise:NSInvalidArgumentException
                    format:@"Cannot specify `ackCallback` block for QoS = 0."];
    }
```